persistence/test/setup_test.go

```diff
@@ -56,84 +40,23 @@ var (
 	DefaultUnstakingHeight = int64(-1)
 )
 
-var testPostgresDB *pgx.Conn
-
 // See https://github.com/ory/dockertest as reference for the template of this code
 // Postgres example can be found here: https://github.com/ory/dockertest/blob/v3/examples/PostgreSQL.md
 func TestMain(m *testing.M) {
-	testPostgresDB = new(pgx.Conn)
-
-	opts := dockertest.RunOptions{
-		Repository: "postgres",
-		Tag:        "12.3",
-		Env: []string{
-			"POSTGRES_USER=" + user,
-			"POSTGRES_PASSWORD=" + password,
-			"POSTGRES_DB=" + db,
-		},
-	}
-
-	defer func() {
-		ctx := context.TODO()
-		testPostgresDB.Close(context.TODO())
-		ctx.Done()
-	}()
-
-	// uses a sensible default on windows (tcp/http) and linux/osx (socket)
-	pool, err := dockertest.NewPool("")
-	if err != nil {
-		log.Fatalf("Could not connect to docker: %s", err)
-	}
-
-	// pulls an image, creates a container based on it and runs it
-	resource, err := pool.RunWithOptions(&opts, func(config *docker.HostConfig) {
-		// set AutoRemove to true so that stopped container goes away by itself
-		config.AutoRemove = true
-		config.RestartPolicy = docker.RestartPolicy{Name: "no"}
-	})
-	if err != nil {
-		log.Fatalf("***Make sure your docker daemon is running!!*** Could not start resource: %s\n", err.Error())
-	}
-
-	// Example: https://github.com/ory/dockertest/blob/v3/examples/PostgreSQL.md
-	// Reasoning: https://github.com/ory/dockertest/blob/v3/examples/PostgreSQL.md
-	hostAndPort := resource.GetHostPort("5432/tcp")
-	databaseUrl := fmt.Sprintf(connStringFormat, user, password, hostAndPort, db)
-
-	log.Println("Connecting to database on url: ", databaseUrl)
-
-	// DOCUMENT: Why do we not call `syscall.SIGTERM` here
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
-	go func() {
-		for sig := range c {
-			log.Printf("exit signal %d received\n", sig)
-			if err := pool.Purge(resource); err != nil {
-				log.Fatalf("could not purge resource: %s", err)
-			}
-		}
-	}()
-
-	resource.Expire(120) // Tell docker to hard kill the container in 120 seconds
+	pool, resource := sharedTest.SetupPostgresDocker()
+	PersistenceModule = sharedTest.PersistenceModule
```

Comment:
Per my other comment, I think these should we lowercased (not exposed outside this module) and prefixed with `test` for explicitness.